### PR TITLE
Add checks and JUnit tests for hexencode/hexdecode

### LINKED ISSUE #66: Add more stringent test cases for module JUnit tests
Some examples would be encoding an empty string, encoding a null bytes, handling invalid characters, invalid number of characters. (Please help to think up of more)

### PR DESCRIPTION
**Pull request by sean-phang**
Closes #66

### CODE REVIEW

File: src/main/java/decodex/modules/hex/HexDecoder.java [hunk 1]
```diff
@@ -27,8 +30,12 @@ public HexDecoder() {
      * @return new Data object representing the decoded input.
      */
     @Override
-    public Data run(Data data) {
-        String inputString = data.toString().replace(" ", "");
+    public Data run(Data data) throws ModuleException {
+        String inputString = data.toString().toLowerCase(Locale.ROOT);
```
Comment (rizemon) [inline src/main/java/decodex/modules/hex/HexDecoder.java:34]: Perhaps discard the `Locale.ROOT`? This is because there are a few other string related method calls that also takes in a locale but we mostly didn't specify it, therefore leaving it to Java to use the default.

File: src/test/java/decodex/modules/hex/HexDecoderTest.java [hunk 2]
```diff
@@ -21,4 +25,31 @@ void run_HexEncoding_expectAllAscii() {
 
         assertEquals(expectedString, result.toString());
     }
+
+    @Test
```
Comment (rizemon) [inline src/test/java/decodex/modules/hex/HexDecoderTest.java:29]: Perhaps add a test case for `"\x00` null character?

File: src/main/java/decodex/modules/hex/HexEncoder.java [hunk 3]
```diff
@@ -33,4 +32,5 @@ public Data run(Data data) {
         String encodedString = encodedStringBuilder.toString().trim();
         return new Data(encodedString);
```
Comment (rizemon) [inline src/main/java/decodex/modules/hex/HexEncoder.java:33]: Perhaps not do a `trim()`? This may accidentally lose some data. (e.g the user may intend to keep the trailing whitespaces)

File: src/main/java/decodex/modules/hex/HexDecoder.java [hunk 4]
```diff
@@ -1,8 +1,11 @@
 package decodex.modules.hex;
 
 import decodex.data.Data;
+import decodex.data.exception.ModuleException;
 import decodex.modules.BaseModule;
 import java.util.Arrays;
+import java.util.Locale;
+import java.util.regex.Pattern;
```
Comment (rizemon) [inline src/main/java/decodex/modules/hex/HexDecoder.java:8]: Perhaps remove the unused `Locale` import?

File: src/main/java/decodex/commands/ListCommand.java [hunk 5]
```diff
@@ -16,6 +16,7 @@ public ListCommand() {
     @Override
     public void run(DataManager dataManager, ModuleManager moduleManager, Ui ui) {
         BaseModule[] modules = moduleManager.getModules();
+        assert modules.length > 0 : "number of modules should be greater than 0";
```
Comment (rizemon) [inline src/main/java/decodex/commands/ListCommand.java:19]: Perhaps capitalize the first letter of "number"?

File: src/main/java/decodex/modules/hex/HexDecoder.java [hunk 6]
```diff
@@ -27,8 +28,12 @@ public HexDecoder() {
...
      */
     @Override
-    public Data run(Data data) {
-        String inputString = data.toString().replace(" ", "");
+    public Data run(Data data) throws ModuleException {
+        String inputString = data.toString().toLowerCase();
+
+        if (!isValidHex(inputString)) {
+            throw new ModuleException("Invalid hexadecimal String");
+        }
```
Comment (rizemon) [inline src/main/java/decodex/modules/hex/HexDecoder.java:36]: Perhaps lowercase the first letter of "String"?

**Review by rizemon — CHANGES_REQUESTED** [review]
A few small typos

**Review by arraysius — APPROVED** [review]
LGTM

**Review by rizemon — APPROVED** [review]
LGTM
